lib/take_away.rb

```diff
@@ -0,0 +1,95 @@
+require_relative 'menu'
+require_relative 'order'
+require_relative 'message'
+
+class TakeAway
+
+  MAX_QUANTITY = 20
+
+  attr_reader :order, :order_hash
+
+  def initialize(filename = 'menu.csv')
+    @filename = filename
+    load_menu_from_the_file(filename)
+    @order_hash = Hash.new 0
+  end
+
+  def read_menu(filename = @filename)
+    load_menu_from_the_file(filename)
+    @menu.print_menu
+  end
+
+  def add_to_order(menu_number, quantity)
+    is_dish_on_the_menu?(menu_number)
+    update_order(menu_number, quantity)
+    puts "#{quantity}x #{@menu_hash[menu_number].name}(s) added to your busket."
+  end
```

Comment:
When you start to see nouns like `order` appear in your method names again and again, that's sometimes a sign that the Single Responsibility Principle is being breached somehow.

I notice that `TakeAway` takes responsibility for the contents of an order — does that seem right to you?